java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/HistoricalLog.java

```diff
@@ -80,14 +85,15 @@ public HistoricalLog(final int limit, final String idStringFormat, Object... arg
    * @param noteFormat {@link String#format} format string that describes the event
    * @param args       for the format string, or nothing if none are required
    */
-  public synchronized void recordEvent(final String noteFormat, Object... args) {
+  @FormatMethod
+  public synchronized void recordEvent(@FormatString final String noteFormat, Object... args) {
     final String note = String.format(noteFormat, args);
     final Event event = new Event(note);
     if (firstEvent == null) {
       firstEvent = event;
     }
     if (history.size() == limit) {
-      history.removeFirst();
+      history.remove(0);
```

Comment:
Is this good with ArrayList? Deque is probably preferable

Reply:
A naive 1-1 conversion without putting that much thought. I will check this.

Reply:
Using `ArrayDeque`